File: SimulationUtils/hs1d/src_python/Hillslope1D/process/tools/boussinesq_tools/BoundaryConditions.py
import numpy as np

class BoundaryConditions(object):
    """
        Class defining boundary conditions in xmin and xmax as S, Q or free
        (imposed head,impose flux or free). This properties are stored in two 2
        valued arrays containing boundary types and boundary values. First
        Boundary condition is located downstream (on the river) and the other
        one is located upstream (on the ridge of the hillslope)
        #######################################################################
        INPUTS :
           boundary_type : a two-valued list containing the types corresponding
           to the two boundary_conditions : either S for a constant stock or
           Q for a forced flowrate
           boundary_value : a two-valued list containing the values
           corresponding to each limit. In m/s for Q
        #######################################################################
    """

    def __init__(self, boundary_type, boundary_value):
        self.boundary_type = boundary_type
        self.boundary_value = boundary_value
        self.edges = np.zeros(shape=(4, 1))
        self.edges_bool = np.zeros(shape=(4, 1))

    def fixed_edge_matrix_boolean(self):
        if self.boundary_type[0] == 'S':
            # Start with a fixed charge S for xmin and forced flux Q for xmax
            self.edges_bool[0] = 1
            self.edges_bool[2] = 1
        elif self.boundary_type[0] == 'Q':
            self.edges_bool[2] = 1

        if self.boundary_type[1] == 'S':
            self.edges_bool[1] = 1
            self.edges_bool[3] = 1
        elif self.boundary_type[1] == 'Q':
            self.edges_bool[3] = 1

        return self.edges_bool

    def fixed_edge_matrix_values(self):
        if self.boundary_type[0] == 'S':
            self.edges[0] = self.boundary_value[0]
        elif self.boundary_type[0] == 'Q':
            self.edges[2] = self.boundary_value[0]

        if self.boundary_type[1] == 'S':
            self.edges[1] = self.boundary_value[1]
        elif self.boundary_type[1] == 'Q':
            self.edges[3] = self.boundary_value[1]

        return self.edges

File: SimulationUtils/hs1d/src_matlab/process/tools/boussinesq_resolution_tools/boundary_conditions.m
classdef boundary_conditions
    properties
        type    % 2 value string array defining the type (S, Q or free) of the boundary conditions for xmin et xmax 
        value   % 2 value array giving the value of the boundary condition (Slim or Qlim)
    end
    
    methods(Access=public)
        function obj=boundary_conditions(type,value)
            obj.type=type;
            obj.value=value;
        end
        
        function Edges=fixed_edge_matrix_boolean(obj)
            Edges=zeros(4,1);
            if(strcmp(obj.type{1},'S'))
                Edges(1)=1;
                Edges(3)=1;
            elseif(strcmp(obj.type{1},'Q'))
                Edges(3)=1;
            end
            if(strcmp(obj.type{2},'S'))
                Edges(2)=1;
                Edges(4)=1;
            elseif(strcmp(obj.type{2},'Q'))
                Edges(4)=1;
            end
        end
        
        function Edges=fixed_edge_matrix_values(obj)
            Edges=zeros(4,1);
            if(strcmp(obj.type{1},'S'))
                Edges(1)=obj.value(1);
            elseif(strcmp(obj.type{1},'Q'))
                Edges(3)=obj.value(1);
            end
            if(strcmp(obj.type{2},'S'))
                Edges(2)=obj.value(2);
            elseif(strcmp(obj.type{2},'Q'))
                Edges(4)=obj.value(2);
            end
        end
    end
end